aws_xray_sdk/core/sampling/rule_poller.py

```diff
@@ -19,7 +19,9 @@ def __init__(self, cache, connector):
         self._connector = connector
 
     def start(self):
-        threading.Thread(target=self._worker).start()
+        poller_thread = threading.Thread(target=self._worker)
+        poller_thread.daemon = True
```

Comment:
Testing to consider:
1) Checking whether pollers start normally on starting application
2) If any exception occurs on any of the pollers, it doesn't break the application and the exception is caught. The poller doesn't die and continues to fetch information on next iteration.
3) The issue this solves the problem, i.e on terminating the application, the poller thread now terminates.

Please let me know if we have expected results.

Reply:
1. Pollers only start if sampling=True and beginSegment/beginSubsegment is called. This has always been the behavior, and the change hasn't affected it from my own tests.
2. Yes, on an exception, it doesn't break the application. In the _worker() method of the pollers, when any exception is caught, it simply logs it and retries the same cycle.
3. I had tested that this fix solves the problem using the minimal configuration shown in the comment I had posted. Prior to the fix, main thread is blocked. After the fix, main thread is no longer blocked and so the poller threads actually terminate.